src/pages/Swap.tsx

```diff
@@ -206,50 +191,81 @@ function Swap(): ReactElement {
         }))
         return
       }
-      // TODO: improve the relationship between token / index
-      const tokenIndexFrom = activePool.tokens.findIndex(
-        ({ symbol }) => symbol === formStateArg.from.symbol,
-      )
-      const tokenIndexTo = activePool.tokens.findIndex(
-        ({ symbol }) => symbol === formStateArg.to.symbol,
-      )
       const amountToGive = parseUnits(
         cleanedFormFromValue,
         TOKENS_MAP[formStateArg.from.symbol].decimals,
       )
+      const tokenFrom = TOKENS_MAP[formStateArg.from.symbol]
+      const tokenTo = TOKENS_MAP[formStateArg.to.symbol]
       let error: string | null = null
-      let amountToReceive: BigNumber
-      if (amountToGive.gt(tokenBalances[formStateArg.from.symbol])) {
+      let amountToReceive = Zero
+      let amountMediumSynth = Zero
+      if (amountToGive.gt(tokenBalances[formStateArg.from.symbol] || Zero)) {
         error = t("insufficientBalance")
       }
       if (amountToGive.isZero()) {
         amountToReceive = Zero
-      } else {
-        amountToReceive = await activePool.contract.calculateSwap(
-          tokenIndexFrom,
-          tokenIndexTo,
+      } else if (formStateArg.swapType === SWAP_TYPES.TOKEN_TO_TOKEN) {
+        const originPool = POOLS_MAP[formStateArg.from.poolName]
+        const destinationPool = POOLS_MAP[formStateArg.to.poolName]
+        const [
+          amountOutSynth,
+          amountOutToken,
+        ] = await bridgeContract.calcTokenToToken(
+          [originPool.addresses[chainId], destinationPool.addresses[chainId]],
+          formStateArg.from.tokenIndex,
+          formStateArg.to.tokenIndex,
+          amountToGive,
+        )
+        amountToReceive = amountOutToken
+        amountMediumSynth = amountOutSynth
+      } else if (formStateArg.swapType === SWAP_TYPES.SYNTH_TO_TOKEN) {
+        const destinationPool = POOLS_MAP[formStateArg.to.poolName]
+        const [
+          amountOutSynth,
+          amountOutToken,
+        ] = await bridgeContract.calcSynthToToken(
+          destinationPool.addresses[chainId],
+          utils.formatBytes32String(formStateArg.from.symbol),
+          formStateArg.to.tokenIndex,
+          amountToGive,
+        )
+        amountToReceive = amountOutToken
+        amountMediumSynth = amountOutSynth
+      } else if (formStateArg.swapType === SWAP_TYPES.TOKEN_TO_SYNTH) {
+        const originPool = POOLS_MAP[formStateArg.from.poolName]
+        amountToReceive = await bridgeContract.calcTokenToSynth(
+          originPool.addresses[chainId],
+          formStateArg.from.tokenIndex,
+          utils.formatBytes32String(formStateArg.to.symbol),
+          amountToGive,
+        )
+      } else if (swapContract != null) {
+        amountToReceive = await swapContract.calculateSwap(
+          formStateArg.from.tokenIndex,
+          formStateArg.to.tokenIndex,
           amountToGive,
         )
       }
```

Comment:
Paremeters to these calls look good to me. Uses correct `formatBytes32String(symbol)` for synths and uses correct indexes for tokens in the pools.